### ISSUE TITLE
Enforcing new rules for plugin:vue/recommended in a minor version, intentional?

### ISSUE THREAD

**Issue by bcruddy** [vuejs/eslint-plugin-vue#563]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **eslint-plugin-vue Version:** 4.7.1 / 4.2.2
* **Node Version:** 8.11.3

**Please show your full configuration:**

```
"extends": [
        "standard",
        "plugin:vue/recommended"
    ],
```

**What did you do? Please include the actual source code causing the issue.**

Ran eslint.

**What did you expect to happen?**

New rules to be "enforced" as warnings between minor versions 4.2.2 and 4.7.1.

Pinning is a fine but troubleshooting this was a PITA. Are breaking changes something we should expect in minor versions to `plugin:vue/recommended`? It seems that these should be warnings instead of errors to prevent this from breaking builds.

**What actually happened? Please include the actual, raw output from ESLint.**

New rules were enforced as errors between minor versions.

```
/path/to/file.vue
  11:21  error  Attribute ":class" should go before "class"  vue/attributes-order
  20:25  error  Attribute ":title" should go before "class"  vue/attributes-order
  26:25  error  Attribute "v-if" should go before "class"    vue/attributes-order
  31:25  error  Attribute "v-if" should go before "class"    vue/attributes-order

/path/to/file.vue
  290:17  warning  Unexpected 'todo' comment  no-warning-comments

/path/to/file.vue
  42:21  error  Attribute ":href" should go before "@click"   vue/attributes-order
  43:21  error  Attribute ":class" should go before "@click"  vue/attributes-order
  44:21  error  Attribute "class" should go before "@click"   vue/attributes-order
  45:21  error  Attribute "target" should go before "@click"  vue/attributes-order

/path/to/file.vue
  10:9  error  Attribute ":group-values" should go before "autocomplete"  vue/attributes-order
  11:9  error  Attribute ":disabled" should go before "autocomplete"      vue/attributes-order
  12:9  error  Attribute ":placeholder" should go before "autocomplete"   vue/attributes-order
  13:9  error  Attribute ":edit-cls" should go before "autocomplete"      vue/attributes-order

/path/to/file.vue
  18:21  error  Attribute "theme" should go before "@onClick"      vue/attributes-order
  19:21  error  Attribute "class" should go before "@onClick"      vue/attributes-order
  20:21  error  Attribute ":disabled" should go before "@onClick"  vue/attributes-order

/path/to/file.vue
   5:13  error  Attribute ":class" should go before "class"        vue/attributes-order
  11:17  error  Attribute ":searchable" should go before "label"   vue/attributes-order
  12:17  error  Attribute ":disabled" should go before "label"     vue/attributes-order
  16:17  error  Attribute "theme" should go before "@onClick"      vue/attributes-order
  17:17  error  Attribute "class" should go before "@onClick"      vue/attributes-order
  18:17  error  Attribute ":disabled" should go before "@onClick"  vue/attributes-order
  24:13  error  Attribute "v-if" should go before "class"          vue/attributes-order
  44:33  error  Attribute ":class" should go before "class"        vue/attributes-order

/path/to/file.vue
  29:21  error  Attribute "theme" should go before "@onClick"  vue/attributes-order

/path/to/file.vue
  18:17  error  Attribute ":value" should go before "@onChange"  vue/attributes-order

/path/to/file.vue
   4:9   error  Attribute ":class" should go before "class"                               vue/attributes-order
  34:17  error  Attribute ":pagination-enabled" should go before "navigation-prev-label"  vue/attributes-order
  35:17  error  Attribute ":per-page" should go before "navigation-prev-label"            vue/attributes-order
  36:17  error  Attribute ":per-page-custom" should go before "navigation-prev-label"     vue/attributes-order
  52:29  error  Attribute "v-if" should go before "class"                                 vue/attributes-order
  56:29  error  Attribute ":disabled" should go before "placement"                        vue/attributes-order
  63:29  error  Attribute "v-if" should go before "class"                                 vue/attributes-order
  67:29  error  Attribute ":disabled" should go before "placement"                        vue/attributes-order
  81:29  error  Attribute "v-if" should go before "class"                                 vue/attributes-order
  85:29  error  Attribute ":disabled" should go before "placement"                        vue/attributes-order

/path/to/file.vue
   11:17  error  Attribute ":disabled" should go before "@click"       vue/attributes-order
   12:17  error  Attribute "class" should go before "@click"           vue/attributes-order
   19:17  error  Attribute ":disabled" should go before "@click"       vue/attributes-order
   20:17  error  Attribute "class" should go before "@click"           vue/attributes-order
   28:21  error  Attribute "kind" should go before "@onClick"          vue/attributes-order
   29:21  error  Attribute "theme" should go before "@onClick"         vue/attributes-order
   34:21  error  Attribute "kind" should go before "@onClick"          vue/attributes-order
   35:21  error  Attribute "theme" should go before "@onClick"         vue/attributes-order
   36:21  error  Attribute "class" should go before "@onClick"         vue/attributes-order
   84:37  error  Attribute "class" should go before "@change"          vue/attributes-order
  107:33  error  Attribute "class" should go before "@onSelect"        vue/attributes-order
  108:33  error  Attribute "placeholder" should go before "@onSelect"  vue/attributes-order
  109:33  error  Attribute "track-by" should go before "@onSelect"     vue/attributes-order
  110:33  error  Attribute "label" should go before "@onSelect"        vue/attributes-order
  133:33  error  Attribute "kind" should go before "@onClick"          vue/attributes-order
  134:33  error  Attribute "theme" should go before "@onClick"         vue/attributes-order
  135:33  error  Attribute "class" should go before "@onClick"         vue/attributes-order
  141:33  error  Attribute "kind" should go before "@onClick"          vue/attributes-order
  142:33  error  Attribute "theme" should go before "@onClick"         vue/attributes-order
  143:33  error  Attribute "class" should go before "@onClick"         vue/attributes-order
  148:33  error  Attribute "id" should go before "@change"             vue/attributes-order
  149:33  error  Attribute "class" should go before "@change"          vue/attributes-order
  150:33  error  Attribute "type" should go before "@change"           vue/attributes-order

/path/to/file.vue
   11:17  error  Attribute ":disabled" should go before "@click"       vue/attributes-order
   12:17  error  Attribute "class" should go before "@click"           vue/attributes-order
   20:21  error  Attribute "kind" should go before "@onClick"          vue/attributes-order
   21:21  error  Attribute "theme" should go before "@onClick"         vue/attributes-order
   26:21  error  Attribute "kind" should go before "@onClick"          vue/attributes-order
   27:21  error  Attribute "theme" should go before "@onClick"         vue/attributes-order
   28:21  error  Attribute "class" should go before "@onClick"         vue/attributes-order
   76:41  error  Attribute "class" should go before "@onSelect"        vue/attributes-order
   77:41  error  Attribute "placeholder" should go before "@onSelect"  vue/attributes-order
   78:41  error  Attribute "track-by" should go before "@onSelect"     vue/attributes-order
   79:41  error  Attribute "label" should go before "@onSelect"        vue/attributes-order
  111:37  error  Attribute ":class" should go before "@change"         vue/attributes-order
  112:37  error  Attribute "class" should go before "@change"          vue/attributes-order
  122:37  error  Attribute "class" should go before "@change"          vue/attributes-order
  177:37  error  Attribute ":class" should go before "@change"         vue/attributes-order
  178:37  error  Attribute "class" should go before "@change"          vue/attributes-order
  186:37  error  Attribute "class" should go before "@change"          vue/attributes-order

/path/to/file.vue
  26:17  error  Attribute ":disabled" should go before "@click"  vue/attributes-order

/path/to/file.vue
  8:13  error  Attribute "v-for" should go before "class"  vue/attributes-order
  9:13  error  Attribute ":key" should go before "class"   vue/attributes-order

/path/to/file.vue
  30:33  error  Attribute ":disabled" should go before "label"  vue/attributes-order

/path/to/file.vue
   8:17  error  Attribute "v-model.number" should go before "type"  vue/attributes-order
   9:17  error  Attribute ":placeholder" should go before "type"    vue/attributes-order
  10:17  error  Attribute ":min" should go before "type"            vue/attributes-order
  11:17  error  Attribute ":max" should go before "type"            vue/attributes-order

/path/to/file.vue
   8:17  error  Attribute "v-model.number" should go before "type"  vue/attributes-order
   9:17  error  Attribute ":min" should go before "type"            vue/attributes-order
  15:13  error  Attribute "track-by" should go before "@input"      vue/attributes-order
  16:13  error  Attribute "label" should go before "@input"         vue/attributes-order
  17:13  error  Attribute ":options" should go before "@input"      vue/attributes-order
  18:13  error  Attribute ":searchable" should go before "@input"   vue/attributes-order

/path/to/file.vue
  5:13  error  Attribute "v-for" should go before "class"  vue/attributes-order
  6:13  error  Attribute ":key" should go before "class"   vue/attributes-order

/path/to/file.vue
  6:13  error  Attribute "v-for" should go before "class"  vue/attributes-order
  7:13  error  Attribute ":key" should go before "class"   vue/attributes-order

/path/to/file.vue
   4:9   error  Attribute "v-if" should go before "id"            vue/attributes-order
  26:29  error  Attribute "v-model" should go before "autofocus"  vue/attributes-order
  37:29  error  Attribute "v-if" should go before "class"         vue/attributes-order
  40:33  error  Attribute ":class" should go before "class"       vue/attributes-order
  45:33  error  Attribute "v-for" should go before "class"        vue/attributes-order
  46:33  error  Attribute ":key" should go before "class"         vue/attributes-order
  50:37  error  Attribute ":class" should go before "class"       vue/attributes-order
  62:29  error  Attribute "v-if" should go before "class"         vue/attributes-order
  69:33  error  Attribute ":disabled" should go before "@click"   vue/attributes-order
  72:37  error  Attribute ":class" should go before "class"       vue/attributes-order
```

**Comment by michalsnik** [comment]
Hey @bcruddy It certainly wasn't intentional. We try to release new rules as `Uncategorized` and put them in right configs only on `major` releases or during `beta` period of the upcoming major release. Sorry for this oversight.
